build/dsc.rake

```diff
@@ -52,96 +52,116 @@ Default values:
-      dsc_resources_path = File.expand_path(dsc_resources_path)
+      dsc_resources_path     = args[:dsc_resources_path] || default_dsc_resources_path
+      dsc_resources_path     = File.expand_path(dsc_resources_path)
       dsc_resources_path_tmp = "#{dsc_resources_path}_tmp"
-      update_versions = args[:update_versions] || false
-      is_custom_resource = (dsc_resources_path != default_dsc_resources_path)
-
+      update_versions        = args[:update_versions] || false
+      is_custom_resource     = (dsc_resources_path != default_dsc_resources_path)
+      
       if !is_custom_resource
         puts "Downloading and Importing #{item_name}"
         cmd = ''
         cmd = "git clone #{dsc_repo} #{dsc_resources_path_tmp} && " unless Dir.exist? dsc_resources_path_tmp
         cmd += "cd #{dsc_resources_path_tmp}"
         cmd += " && git checkout #{ENV['DSC_REF']}" if ENV['DSC_REF']
         cmd += " && git submodule update --init"
-
         sh cmd
-      else
-        puts "Importing custom types from '#{dsc_resources_path}'"
-        FileUtils.mkdir_p "#{dsc_resources_path_tmp}"
-        FileUtils.cp_r "#{dsc_resources_path}/.", "#{dsc_resources_path_tmp}/"
-      end
 
-      composite_resources = [ 'xChrome','xDSCResourceDesigner','xDscDiagnostics',
-        'xFirefox','xSafeHarbor','xSystemSecurity' ]
-
-      Rake::Task['dsc:resources:checkout'].invoke(
-        "#{dsc_resources_path_tmp}/xDscResources", update_versions, composite_resources)
-      Rake::Task['dsc:resources:checkout'].invoke(
-        "#{dsc_resources_path_tmp}/dscresources", update_versions, composite_resources)
-
-      # filter out unwanted files
-      valid_files = Dir.glob("#{dsc_resources_path_tmp}/**/*").reject do |f|
-        # reject the .git folder or special git files
-        f =~ /\/\.(git|gitattributes|gitignore|gitmodules)/ ||
-        # reject binary and other file extensions
-        f =~ /\.(pptx|docx|sln|cmd|xml|pssproj|pfx|html|txt|xlsm|csv|png|git|yml|md)$/i ||
-        # reject test / sample / example code
-        f =~ /\/.*([Ss]ample|[Ee]xample|[Tt]est).*/ ||
-        # reject stuff that is a Composite DSC Resource
-        f =~ /(xChrome|xDSCResourceDesigner|xDscDiagnostics|xFirefox|xSafeHarbor|xSystemSecurity).*/ ||
-        # reject duplicated resources
-        f =~ /(xSharePoint|PSDscResources).*/ ||
-        # and don't keep track of dirs
-        Dir.exists?(f)
-      end
+        composite_resources = [ 'xChrome','xDSCResourceDesigner','xDscDiagnostics',
+          'xFirefox','xSafeHarbor','xSystemSecurity' ]
+
+        Rake::Task['dsc:resources:checkout'].invoke(
+          "#{dsc_resources_path_tmp}/xDscResources", update_versions, composite_resources)
+        Rake::Task['dsc:resources:checkout'].invoke(
+          "#{dsc_resources_path_tmp}/dscresources", update_versions, composite_resources)
+
+        # filter out unwanted files
+        valid_files = Dir.glob("#{dsc_resources_path_tmp}/**/*").reject do |f|
```

Comment:
Since this code looks to now be duplicated.. can we add a single commit to refactor it to a method? (preferably as the first commit in the PR)